File: dungeons-1.21.11/src/main/java/fishmod/utils/rendering/RenderHandler.java
package fishmod.utils.rendering;

import java.util.ArrayList;
import java.util.List;
import java.util.function.Consumer;

public class RenderHandler {

    private final List<RenderingEvent> listeners = new ArrayList<>();

    public void register(RenderingEvent listener) {
        listeners.add(listener);
    }

    public void invoke(Consumer<RenderingEvent> action) {
        if (listeners.isEmpty()) return;
        for (RenderingEvent listener : listeners) {
            action.accept(listener);
        }
    }

}

File: dungeons-26.1.2/src/main/kotlin/fishmod/utils/rendering/RenderHandler.kt
package fishmod.utils.rendering

import java.util.function.Consumer

class RenderHandler {

    private val listeners = ArrayList<RenderingEvent>()

    fun register(listener: RenderingEvent) {
        listeners.add(listener)
    }

    fun invoke(action: Consumer<RenderingEvent>) {
        if (listeners.isEmpty()) return
        for (listener in listeners) {
            action.accept(listener)
        }
    }

}
